Move login to redux

neilff opened the pull request:
I added a promise middleware for resolving the response from the mock API. Show / hiding the error is now based on the app state and not local component state.

I'm not sure if this is taking the seed too far though..

neilff commented:
@andrejkn @rbrander can you take a look at these potential changes and provide feedback?

rbrander commented on src/utils/utilities.js:
```diff
@@ -16,3 +16,15 @@ export function immutableToJS(state) {
+ * @return {Boolean}
+ */
+export function isPromise(value) {
+  if (value !== null && typeof value === 'object') {
```
it might be a good idea to add a unit test for this

rbrander commented:
The code looks great overall.  I'd recommend you squash your commits before merging.  Other than that, LGTM. :shipit:

neilff commented:
@rbrander, @andrejkn 
- Added unit tests
- Added timeout for login mock, simulate a delayed api

Still need to do some more work, but will submit as additional PRs after this is merged

andrejkn commented:
@neilff looks good :shipit:

neilff commented:
I actually don't have the ability to merge, feel free to do so once you are cool with it :+1:

andrejkn commented:
@rbrander we need WRITE permission :)
